spec/front_desk_spec.rb

```diff
@@ -0,0 +1,398 @@
+      end
+
+      it "will say element at all_rooms index 0 is 1" do
+        expect(@concierge.all_rooms_array[0]).must_equal 1
+      end
+      it "will say element at all_rooms index 19 is 20" do
+        expect(@concierge.all_rooms_array[19]).must_equal 20
+      end
+
+      it "will say reservations_record is an array" do
+        expect(@concierge.reservations_record).must_be_kind_of Array
+      end
+
+      it "will say block_reservations_only is an array " do
+        expect(@concierge.block_reservations_only).must_be_kind_of Array
+      end
+
+      it "will say reserved_rooms_in_blocks_is an array" do
+        expect(@concierge.reserved_rooms_in_blocks).must_be_kind_of Array
+      end
+
+      it "will say reserved_rooms_in_blocks array length is 0" do
+        expect(@concierge.reserved_rooms_in_blocks.length).must_equal 0
+      end
+
+      it "will say block_reservations_only array length is 0" do
+        expect(@concierge.block_reservations_only.length).must_equal 0
+      end
+
+      it "will say reservations_record array length is 0" do
+        expect(@concierge.reservations_record.length).must_equal 0
+      end
+    end
+
+    describe "make_reservation for block and add_reservation methods" do
+      before do
+        @concierge = build_front_desk
+      end
+      it "will describe various attributes of the reservation" do
+        reservation_1 = @concierge.make_reservation(
+          1201,
+          5,
+          1,
+          Date.new(2019, 3, 20),
+          Date.new(2019, 3, 24),
+        )
+        @concierge.add_reservation(reservation_1)
+        expect(reservation_1.booking_ref).must_equal 1201
+        expect(reservation_1.number_of_rooms).must_equal 5
+        expect(reservation_1.first_room_number).must_equal 1
+        expect(@concierge.reservations_record.length).must_equal 1
+        expect(@concierge.reservations_record[0].total_cost).must_equal 3200.00
+        expect(@concierge.reservations_record[0].room_blocks[4]).must_equal 5
+        expect(@concierge.reservations_record[0].room_blocks.length).must_equal 5
+        expect(reservation_1.check_in).must_equal Date.new(2019, 3, 20)
+        expect(reservation_1.check_out).must_equal Date.new(2019, 3, 24)
+      end
+    end
+
+    describe "get reservation by a specific date" do
+      before do
+        @concierge = build_front_desk
+      end
+
+      it "will get a reservation by a date" do
+        reservation_2 = @concierge.make_reservation(
+          1202,
+          2,
+          9,
+          Date.new(2019, 4, 8),
+          Date.new(2019, 4, 10),
+        )
+
+        @concierge.add_reservation(reservation_2)
+
+        by_date = @concierge.get_reservations_by_date(Date.new(2019, 4, 8))
+
+        expect(by_date).must_be_kind_of Array
+        expect(by_date.length).must_equal 1
+        expect(reservation_2.check_in).must_be :<, Date.new(2019, 4, 10)
+        expect(reservation_2.check_in).must_be :>, Date.new(2019, 4, 7)
```

Comment:
You probably don't need to make assertions on `reservation2` in this test, since you didn't alter it